referenceqvm/unitary_generator.py

```diff
@@ -355,6 +356,45 @@ def tensor_gates(gate_set, defgate_set, pyquil_gate, num_qubits):
     return gate
 
 
+def tensor_up(pauli_terms, num_qubits):
+    """
+    Takes a PauliSum object along with a total number of
+    qubits and returns a matrix corresponding the tensor representation of the
+    object.
+
+    Useful for generating the full Hamiltonian after a particular fermion to
+    pauli transformation.
+
+    :param pauli_terms: (PauliSum) object of PauliTerm
+    :param num_qubits: (int) number of qubits in the system
+    :returns: (numpy array) representation of the paui_terms operator
+    """
+    if not isinstance(pauli_terms, PauliSum):
+        raise TypeError("can only tensor PauliSum")
+
+    if __debug__:
+        for term in pauli_terms.terms:
+            if len(term._ops.keys()) > 0:
+                if max(term._ops.keys()) >= num_qubits:
+                    raise IndexError("pauli_terms has higher index than qubits")
+
+    big_hilbert = np.zeros((2 ** num_qubits, 2 ** num_qubits))
+    # left kronecker product corresponds to the correct basis ordering
+    for term in pauli_terms.terms:
+        tmp_big_hilbert = np.array([1])
```

Comment:
move this in front of the for loop to visually indicate that they belong together and form a `fold` operation.
moreover this assignment is too heavy. A simple
```tmp_big_hilbert = 1``` will do the trick

Reply:
The `np.kron` routine uses np.asarray on both inputs to cast as an array object.  Just being explicit about the type of object that I am performing the tensor product with--i.e. both objects are `numpy.ndarray`.